opslevel/resource_opslevel_property_assignment.go

```diff
@@ -1,113 +1,174 @@
+					JsonStringValidator(),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+func (resource *PropertyAssignmentResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var planModel PropertyAssignmentResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	definition := planModel.Definition.ValueString()
+	owner := planModel.Owner.ValueString()
+	value := opslevel.JsonString(planModel.Value.ValueString())
+	input := opslevel.PropertyInput{
+		Definition: *opslevel.NewIdentifier(planModel.Definition.ValueString()),
+		Owner:      *opslevel.NewIdentifier(planModel.Owner.ValueString()),
+		Value:      value,
+	}
+	assignment, err := resource.client.PropertyAssign(input)
+	if err != nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("failed to assign property (%s) on service (%s), got error: %s", definition, owner, err))
+		return
+	}
+
+	stateModel := NewPropertyAssignmentResourceModel(*assignment)
+	stateModel.LastUpdated = timeLastUpdated()
+	// user is free to use either alias or ID for 'owner' and 'definition' fields
+	stateModel.Owner = planModel.Owner
+	stateModel.Definition = planModel.Definition
+
+	tflog.Trace(ctx, fmt.Sprintf("assigned property (%s) on service (%s) with value: '%s'", definition, owner, value))
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+}
+
+func (resource *PropertyAssignmentResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var planModel PropertyAssignmentResourceModel
+	resp.Diagnostics.Append(req.State.Get(ctx, &planModel)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	definition := planModel.Definition.ValueString()
+	owner := planModel.Owner.ValueString()
+	assignment, err := resource.client.GetProperty(owner, definition)
+	if err != nil || assignment == nil {
+		resp.Diagnostics.AddError("opslevel client error", fmt.Sprintf("unable to read property assignment (%s) on service (%s), got error: %s", definition, owner, err))
+		return
+	}
+	value := *assignment.Value
+
+	stateModel := NewPropertyAssignmentResourceModel(*assignment)
+	// user is free to use either alias or ID for 'owner' and 'definition' fields
+	stateModel.Owner = planModel.Owner
+	stateModel.Definition = planModel.Definition
+
+	tflog.Trace(ctx, fmt.Sprintf("read property assignment (%s) on service (%s) with value: '%s'", definition, owner, value))
+	resp.Diagnostics.Append(resp.State.Set(ctx, &stateModel)...)
+}
+
+func (resource *PropertyAssignmentResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	resp.Diagnostics.AddError("terraform plugin error", "property assignments should never be updated, only replaced.\nplease file a bug report including your .tf file at: github.com/OpsLevel/terraform-provider-opslevel")
```

Comment:
Just like the old provider, update() should never happen because of ForceNew()/RequiresReplace()